src/main/java/org/spongepowered/api/data/DataManager.java

```diff
@@ -186,4 +187,22 @@
     <T extends DataManipulator<T, I>, I extends ImmutableDataManipulator<I, T>>
         Optional<DataManipulatorBuilder<T, I>> getImmutableManipulatorBuilder(Class<I> immutableManipulatorClass);
 
+    /**
+     * Registers a {@link DataSerializer} for the desired class.
+     *
+     * @param serializer The serializer for the desired class object
+     * @param objectClass The class of the object type being managed
+     * @param <T> The type of object
+     */
+    <T> void registerSerializer(DataSerializer<T> serializer, Class<T> objectClass);
```

Comment:
Flip the arguments? It makes more sense to do `manager.registerSerializer(MyClass.class, new DataSerializer.....);`.